### PR TITLE
feat: get beehiiv posts

### PR DESCRIPTION
**Pull request by preschian**
related https://github.com/kodadot/nft-gallery/issues/8423

### CODE REVIEW

**Comment by yangwao** [comment]
let's merge and deploy

File: newsletter/src/index.ts [hunk 1]
```diff
@@ -63,4 +63,16 @@ app.put('/subscribe/resend-confirmation', subscribeValidator, async (c) => {
 	return c.json(undefined, 204);
 });
 
+app.get('/index', async (c) => {
```
Comment (roiLeo) [inline newsletter/src/index.ts:66]: `/blogs` or `/posts`?

File: newsletter/src/utils/beehiiv.ts [hunk 2]
```diff
@@ -42,3 +42,19 @@ export const deleteSubscription = (subscriptionId: string, c: Context<HonoEnv>)
...
 };
+
+export const indexPosts = (c: Context<HonoEnv>) => {
+	const apiKey = c.env.BEEHIIV_API_KEY;
+	const publicationId = c.env.BEEHIIV_PUBLICATION_ID;
+
+	const beehiiv = new URL(BEEHIIV_API_URL);
+	beehiiv.pathname += `/publications/${publicationId}/posts`;
+	beehiiv.searchParams.append('status', 'confirmed');
+	beehiiv.searchParams.append('direction', 'desc');
```
Comment (roiLeo) [inline newsletter/src/utils/beehiiv.ts:53]: maybe pass sorting option?

**Review by roiLeo — APPROVED** [review]

**Comment by yangwao** [comment]
@preschian let's resolve conflicts and we can merge this one

**Comment by yangwao** [comment]
ok random hit

**Pull request merged**